server/app/controllers/HomeController.java

```diff
@@ -49,15 +48,7 @@ public HomeController(
   }
 
   public CompletionStage<Result> index(Http.Request request) {
-    Optional<CiviFormProfile> maybeProfile = profileUtils.currentUserProfile(request);
-
-    // If the user isn't already logged in within their browser session, consider them a guest.
-    if (maybeProfile.isEmpty()) {
-      return CompletableFuture.completedFuture(createGuestSessionAndRedirect(request));
```

Comment:
**thought:** Same here - I wonder if its worth putting in a change to add some logging here to make sure this case doesn't happen then follow up with this change.

Reply:
I think that might be overkill here - the filter has simple logic to determine whether it runs (is a GET request, is not an excluded path) so a simple test shows that it always runs for this controller. Is there some other risk I'm not thinking about, and/or do you feel strongly that this a metric here is necessary to mitigate possible unknown risks?

Reply:
For extra context, I'm working on another PR https://github.com/civiform/civiform/pull/8443 that makes this same change in the rest of the user-facing controllers - I separated this one out because of the required changes in the filter.

Reply:
Actually I changed this PR to include the only two controllers that explicitly have the redirect; so that linked PR (which I might abandon) will actually just address code that made the profile optional when it probably doesn't need to be anymore.

Reply:
Sounds good, thanks! I can't think of any exact risks with this.